molecule/templates/vagrantfile.j2

```diff
@@ -3,6 +3,9 @@ Vagrant.configure('2') do |config|
   {%- for platform in config.vagrant.platforms %}
     {%- if platform.name == current_platform %}
   config.vm.box = '{{ platform.box }}'
+      {%- if 'box_version' in platform %}
+  config.vm.box_version = "{{ platform.box_version|join }}"
```

Comment:
box_version is optional, if it exists in molecule.yml, we render it in vagrantfile